### PR TITLE
Bluetooth: Mesh: make models metadata const

### LINKED ISSUE #69684: RFC: API Change: BLE Mesh (access)
### Problem description

https://github.com/zephyrproject-rtos/zephyr/pull/69679

Bluetooth Mesh specification claims:
```
4.2.50.1 Models Metadata Page 0
Composition Metadata Page 0 shall not change during a term of a node on the network (see Section
4.2.2.1).
```

as well as Metadata Composition Page 0 is a part of Composition Data Page 0 that is:

```
4.2.2.1 Composition Data Page 0
Composition Data Page 0 shall be present on a node. Composition Data Page 0 shall not change during
a term of a node on the network.
```

However, absence of `const` qualifier even does not allow making Models Metadata state permanent.
If there are a couple o models those have metadata states, it causes unnecessary RAM consumption.

If API user declares Models Metadata instances with `const` qualifier by self to save memory and follow specification, he\she has to suppress the fair enough warnings about discarded qualifiers.

### Proposed change

Add 'const' qualifier to metadata pointers `bt_mesh_models_metadata_entry **metadata` within model structure `struct bt_mesh_model`.

### Dependencies

The Health server `struct bt_mesh_health_srv` metadata state pointer `struct bt_mesh_models_metadata_entry *metadata` gets `const` qualifier too.

### Concerns and Unresolved Questions

Since change adds qualifier, then the API user will not get any warning in the current implementations. No potential negative impact.
Models Metadata state is optional and available only if the Large Composition Data server is present that is disabled by default.

### PR DESCRIPTION
**Pull request by alxelax**
Commit adds const qualifier to models metadata.
Specification claims: Composition Metadata Page 0
shall not change during a term of a node on the network.

### CODE REVIEW

**Comment by Thalley** [comment]
Isn't changing the type of a public field a breaking API change? 

https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-breaking-api-changes

**Comment by alxelax** [comment]
> Isn't changing the type of a public field a breaking API change?
> 
> https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-breaking-api-changes

Technically yes. Nevertheless, the user will not get even warning since PR adds `const` qualifier but does not discard.
@Thalley, should I go over Stable API changing procedure?

**Comment by Thalley** [comment]
> Technically yes. Nevertheless, the user will not get even warning since PR adds `const` qualifier but does not discard.

Hmm, you might be right that this is a non-issue :) Feel free to disregard my previous comment

**Review by Andrewpini — DISMISSED** [review]
Seems legit 👍

**Review by PavelVPV — CHANGES_REQUESTED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by PavelVPV — COMMENTED** [review]
LGTM. Please add a note to the migration guide.

**Review by PavelVPV — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

File: doc/releases/migration-guide-3.7.rst [hunk 1]
```diff
@@ -116,6 +116,11 @@ Bluetooth
 Bluetooth Mesh
 ==============
 
+* The model metadata pointer declaration of :c:struct:`bt_mesh_model` has been changed
+  to add `const` qualifiers. The data pointer of :c:struct:`bt_mesh_models_metadata_entry`
+  got `const` qualifier too. The model's metadata structure and metadata raw value
```
Comment (kartben) [inline doc/releases/migration-guide-3.7.rst:121]:
```suggestion
  to add ``const`` qualifiers. The data pointer of :c:struct:`bt_mesh_models_metadata_entry`
  got ``const`` qualifier too. The model's metadata structure and metadata raw value
```

**Review by kartben — CHANGES_REQUESTED** [review]
small formatting issue. Thanks!

**Review by jhedberg — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

File: doc/releases/migration-guide-3.7.rst (hunk 1 above)
Comment (alxelax) [reply doc/releases/migration-guide-3.7.rst]: Hi @kartben, could you take a look at the changes? Are they Ok now?

**Review by kartben — APPROVED** [review]

**Pull request merged**